src/types/dict.rs

```diff
@@ -166,7 +172,7 @@ impl<'a> Iterator for PyDictIterator<'a> {
             let mut key: *mut ffi::PyObject = mem::uninitialized();
             let mut value: *mut ffi::PyObject = mem::uninitialized();
             if ffi::PyDict_Next(self.dict.as_ptr(), &mut self.pos, &mut key, &mut value) != 0 {
-                let py = self.dict.py();
+                let py = Python::assume_gil_acquired();
```

Comment:
I prefer the old version here since it makes sure that this method isn't changed into some gil-less variant one day and this becomes essentially UB

Reply:
`PyObject` doesn't implement `PyObjectWithGIL`.
We need to change this line if we use `PyObject`

Reply:
ok that's fine; Could you add that as a comment for future reference?

Reply:
Now I changed `PyDictIterator` to have `Python`.
Still comment is needed?

Reply:
No, that's clear now